src/qt/walletmodel.cpp

```diff
@@ -571,6 +583,11 @@ bool WalletModel::privateKeysDisabled() const
     return m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
 }
 
+bool WalletModel::canGetAddresses() const
+{
+    return m_wallet->hdEnabled() || (!m_wallet->hdEnabled() && !m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS));
```

Comment:
Suggested comment, since `hdEnabled()` changes in related pull requests, which can get confusing fast:

```cpp
// The wallet can provide a fresh address if:
// * hdEnabled(): an HD seed is present; or
// * it is a legacy wallet, because:
//     * !hdEnabled(): an HD seed is not present; and
//     * private keys have not been disabled (which results in hdEnabled() == true)
```